airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -186,7 +195,13 @@ def _prepare_cli_cmd(self) -> list[Any]:
         return [hive_bin, *cmd_extra, *hive_params_list]
 
     def _validate_beeline_parameters(self, conn):
-        if ":" in conn.host or "/" in conn.host or ";" in conn.host:
+        if self.high_availability:
```

Comment:
This one is somewhat of a security feature - is this expected that `schema` for HA can contain `:` or `/` ? Some examples? I believe all the zookeper etc. parameters will be added anyway above, so there is no particular need to have `/` or `:` in the schema?

Reply:
The reason this was done was for a refactor from earlier. The `schema` still cannot contain `;`

Earlier it was like this:
```
    def _validate_beeline_parameters(self, conn):
        if ":" in conn.host or "/" in conn.host or ";" in conn.host:
            raise Exception(
                f"The host used in beeline command ({conn.host}) should not contain ':/;' characters)"
            )
        try:
            int_port = int(conn.port)
            if not 0 < int_port <= 65535:
                raise Exception(f"The port used in beeline command ({conn.port}) should be in range 0-65535)")
        except (ValueError, TypeError) as e:
            raise Exception(f"The port used in beeline command ({conn.port}) should be a valid integer: {e})")
        if ";" in conn.schema:
            raise Exception(
                f"The schema used in beeline command ({conn.schema}) should not contain ';' character)"
            )
```

To accommodate this logic:
1. Do not validate port when HA is given, because host will be of form: `host1:port1, host2:port2...`
2. We need to validate condition where `;` should not be present even for HA connections. 

So basically it became:
    def _validate_beeline_parameters(self, conn):
        **if self.high_availability:
            if ";" in conn.schema:
                raise Exception(
                    f"The schema used in beeline command ({conn.schema}) should not contain ';' character)"
                )
            return
        elif** ":" in conn.host or "/" in conn.host or ";" in conn.host:
            raise Exception(
                f"The host used in beeline command ({conn.host}) should not contain ':/;' characters)"
            )
        try:
            int_port = int(conn.port)
            if not 0 < int_port <= 65535:
                raise Exception(f"The port used in beeline command ({conn.port}) should be in range 0-65535)")
        except (ValueError, TypeError) as e:
            raise Exception(f"The port used in beeline command ({conn.port}) should be a valid integer: {e})")
        if ";" in conn.schema:
            raise Exception(
                f"The schema used in beeline command ({conn.schema}) should not contain ';' character)"
            )

(Highlighted code is the delta)
@potiuk